pkg/controller/operators/catalog/step.go

```diff
@@ -132,27 +132,33 @@ func (b *builder) NewCRDV1Step(client apiextensionsv1client.ApiextensionsV1Inter
 
 			_, createError := client.CustomResourceDefinitions().Create(context.TODO(), crd, metav1.CreateOptions{})
 			if k8serrors.IsAlreadyExists(createError) {
-				currentCRD, _ := client.CustomResourceDefinitions().Get(context.TODO(), crd.GetName(), metav1.GetOptions{})
-				crd.SetResourceVersion(currentCRD.GetResourceVersion())
-				if err = validateV1CRDCompatibility(b.dynamicClient, currentCRD, crd); err != nil {
-					return v1alpha1.StepStatusUnknown, errors.Wrapf(err, "error validating existing CRs against new CRD's schema: %s", step.Resource.Name)
-				}
+				err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					currentCRD, _ := client.CustomResourceDefinitions().Get(context.TODO(), crd.GetName(), metav1.GetOptions{})
+					crd.SetResourceVersion(currentCRD.GetResourceVersion())
+					if err = validateV1CRDCompatibility(b.dynamicClient, currentCRD, crd); err != nil {
+						return fmt.Errorf("error validating existing CRs against new CRD's schema for %q: %w", step.Resource.Name, err)
+					}
 
-				// check to see if stored versions changed and whether the upgrade could cause potential data loss
-				safe, err := crdlib.SafeStorageVersionUpgrade(currentCRD, crd)
-				if !safe {
-					b.logger.Errorf("risk of data loss updating %s: %s", step.Resource.Name, err)
-					return v1alpha1.StepStatusUnknown, errors.Wrapf(err, "risk of data loss updating %s", step.Resource.Name)
-				}
-				if err != nil {
-					return v1alpha1.StepStatusUnknown, errors.Wrapf(err, "checking CRD for potential data loss updating %s", step.Resource.Name)
-				}
+					// check to see if stored versions changed and whether the upgrade could cause potential data loss
+					safe, err := crdlib.SafeStorageVersionUpgrade(currentCRD, crd)
```

Comment:
Does it make sense to retry only the portion of the codepath that actually gets/updates CRDs and not the business logic around validation that we provide as well? I know right now it's all tangled together. But from what I gather from this change, if there is an conflict error updating the CRD, we retry the get and validation again as well, for no real benefit

Reply:
I thought about that, but since `currentCRD` can change between retries and its an input into the validation, I figured we need to re-run the validation on each retry in case something relevant to the validation also changed.

Reply:
Right, I suppose that is possible